### PR TITLE
docs: adds documentation for Storage Version Migration

### PR DESCRIPTION
**Pull request by drewhagen**
Ref: https://github.com/kubernetes/enhancements/issues/4192
PR - https://github.com/kubernetes/kubernetes/pull/123344

Replaces: https://github.com/kubernetes/website/pull/45294
Considering contributor out of office. This also accounts for review suggestions made by @sftim

x-post on exception thread: https://kubernetes.slack.com/archives/C2C40FMNF/p1711552644986199

/cc @nilekhc @deads2k @jpbetz @enj

/milestone 1.30
/sig docs
/sig api-machinery

### CODE REVIEW

**Comment by drewhagen** [comment]
@kubernetes/sig-docs-en-reviews Can you take another look at this and let me know if you have more feedback that I can iterate on? Thank you! 🙏

**Comment by reylejano** [comment]
This PR replaces PR https://github.com/kubernetes/website/pull/45294 in which the author has not addressed comments as the 1.30 Docs freeze deadline passed. This PR addresses those comments

The original PR has a technical review in this [comment](https://github.com/kubernetes/website/pull/45294#issuecomment-1995342586)

/approve

**Comment by windsonsea** [comment]
Some nits about bullets indentations can be improved later

/lgtm

File: content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    - type: Succeeded
+      status: 'True'
+      lastUpdateTime: '2024-03-12T20:29:46Z'
+      reason: StorageVersionMigrationSucceeded
+    resourceVersion: '84'
+  ```
+- [Verify](/docs/tasks/administer-cluster/kms-provider/#verifying-that-the-data-is-encrypted)
+  the stored secret is now prefixed with `k8s:enc:aescbc:v1:key2`.
+
+## Update the preferred storage schema of the resource while moving from _v1_ to _v2_
```
Comment (tengqm) [inline content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md:128]: How about this?

```suggestion
## Update the preferred storage schema of a CRD
```

File: content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+          caBundle: <CABundle info>
+      conversionReviewVersions:
+      - v1
+      - v2
+  ```
+  Create CRD using kubectl
+  ```shell
+  kubectl apply -f test-crd.yaml
+  ```
+- Create a manifest for an example testcrd. Name the manifest `cr1.yaml` and use these contents:
```
Comment (tengqm) [inline content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md:176]: nit: what is 'testcrd'? this spelling may confuse users.

File: content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+  ```
+  Create CR using kubectl
+  ```shell
+  kubectl apply -f cr1.yaml
+  ```
+- Verify that CR is written and stored as v1 by getting the object from etcd.
+  ```shell
+  ETCDCTL_API=3 etcdctl get /kubernetes.io/stable.example.com/testcrds/default/cr1 [...] | hexdump -C
+  ```
+  where [...] contains the additional arguments for connecting to the etcd server.
```
Comment (tengqm) [inline content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md:192]:
```suggestion
  where `[...]` contains the additional arguments for connecting to the etcd server.
```

File: content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+      group: stable.example.com
+      version: v1
+      resource: SelfieRequest
+  ```
+  Create the object using _kubectl_ as follows:
+  ```shell
+  kubectl apply -f migrate-crd.yaml
+  ```
+- Monitor migration of secrets using status. Successful migration should have
+  _Succeeded_ condition set to _true_ in the status field. Get the migration resource
```
Comment (tengqm) [inline content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md:277]:
```suggestion
  the `Succeeded` condition set to "True" in the status field. Get the migration resource
```

**Pull request merged**